# * Change documentation location
# * Instead of allowing multiple definitions (allocations in header files),
#   declare global variables in headers, and allocate them in their
#   corresponding .c files, with a reasonable initial value.

diff -u3 aris-2.2/src/app.c aris-2.2-patched/src/app.c
--- aris-2.2/src/app.c	2014-02-28 10:17:24.000000000 -0300
+++ aris-2.2-patched/src/app.c	2022-08-11 15:28:59.318566050 -0300
@@ -46,6 +46,9 @@
 #include "elm-conn.xpm"
 #include "nil-conn.xpm"
 
+
+aris_app * the_app = 0;
+
 // Macro to get the length of a file.
 
 #ifndef WIN32
diff -u3 aris-2.2/src/app.h aris-2.2-patched/src/app.h
--- aris-2.2/src/app.h	2014-02-27 22:32:07.000000000 -0300
+++ aris-2.2-patched/src/app.h	2022-08-11 13:01:06.315136621 -0300
@@ -106,6 +106,6 @@
 
 int app_set_color (aris_app * app, int index, int red, int green, int blue);
 /* The main application */
-aris_app * the_app;
+extern aris_app * the_app;
 
 #endif /* ARIS_APP_H */
diff -u3 aris-2.2/src/aris.c aris-2.2-patched/src/aris.c
--- aris-2.2/src/aris.c	2014-02-28 10:29:10.000000000 -0300
+++ aris-2.2-patched/src/aris.c	2022-08-11 15:38:22.990593330 -0300
@@ -844,8 +844,7 @@
 
       if (the_app->working_dir)
 	{
-	  parent = g_file_get_parent (parent);
-	  sprintf (the_app->help_file, "file://%s/doc/aris/index.html", g_file_get_path (parent));
+	  sprintf (the_app->help_file, "file:///usr/doc/aris-" PACKAGE_VERSION "/html/index.html");
 	}
 
       int ret;
diff -u3 aris-2.2/src/process.c aris-2.2-patched/src/process.c
--- aris-2.2/src/process.c	2014-03-06 17:53:32.000000000 -0300
+++ aris-2.2-patched/src/process.c	2022-08-11 15:27:56.582563013 -0300
@@ -26,6 +26,8 @@
 #include <ctype.h>
 #include <math.h>
 
+struct connectives_list main_conns = { 0 };
+
 /* Eliminates a negation from a string.
  *  input:
  *    not_str - the string from which to eliminate the negation.
diff -u3 aris-2.2/src/process.h aris-2.2-patched/src/process.h
--- aris-2.2/src/process.h	2014-03-02 16:30:39.000000000 -0300
+++ aris-2.2-patched/src/process.h	2022-08-11 13:04:45.004147205 -0300
@@ -170,7 +170,7 @@
 
 // The main connectives.
 
-struct connectives_list main_conns;
+extern struct connectives_list main_conns;
 
 // Definitions.
 
